stubs/CoreGenericIterators.phpstub

```diff
@@ -581,8 +586,9 @@ class LimitIterator extends IteratorIterator implements OuterIterator {
 /**
  * @template-covariant TKey
  * @template-covariant TValue
+ * @template TIterator as Traversable<TKey, TValue>
  *
- * @template-extends IteratorIterator<TKey, TValue>
+ * @template-extends IteratorIterator<TKey, TValue, TIterator>
  */
 class NoRewindIterator extends IteratorIterator {
```

Comment:
Should constructor be typed to accept TIterator instead (this applies to all `IteratorIterator` descendants actually)?